salt/docker/init.sls

```diff
@@ -40,21 +41,16 @@ docker:
     - append_if_not_found: True
     - require:
       - pkg: docker
-  # [inercia] when dockerd was already running and we require the
-  # service to be "service.running", Salt does not think it must
-  # restart it even when we say "watch these files", so we
-  # need this "cmd.run"...
-  cmd.run:
+  cmd.wait:
     - name: systemctl restart docker.service
-    - onlyif: systemctl status docker.service
+    - watch:
+      - file: /etc/systemd/system/docker.service.d/proxy.conf
```

Comment:
Last time it tried it didn't. Docker will not be restarted if 1) a fresh installation happens and 2) the docker daemon is enabled on bootup.

In this case, removing a `systemctl status` will not help us make things idempotent and, in fact, I think that is the key for reaching that goal.